Automattic/wp-calypso
# Add tests for hasSupportingPlan functions

### Changes proposed in this Pull Request

In 3 different places in the code, whether a site has the SEO feature or not is checked with the `hasSupportingPlan` function. The offer reset highlighted that checking features directly (e.g. with `planHasFeature`) instead of plans will allow us to more easily add plans in the future.

Since I couldn't find how to check the components that use `hasSupportingPlan` directly in the UI, I decided to first add tests to capture its behaviour. The next step, in a different PR, will be to change its implementation and check for features instead of plans.

### Implementation notes

Code and tests are voluntarily duplicated. In a next PR, all instances of `hasSupportingPlan` will be replaced by a shared function.

### Testing instructions

- Download the PR
- Review the code
- Run `yarn test-client:watch client/post-editor/editor-drawer/test/utils.js` and check all tests pass
- Run `yarn test-client:watch client/my-sites/site-settings/seo-settings/test/utils.js` and check all tests pass
- Run `yarn test-client:watch client/my-sites/site-settings/seo-settings/test/form.jsx` and check all tests pass
- Run `yarn test-client:watch client/components/seo-preview-pane/test/utils.js` and check all tests pass
